app/models/certificate_authority.rb

```diff
@@ -40,4 +39,13 @@ def self.revoked?(key_id, serial)
     # Cert should exist and not be revoked.
     cert&.revoked?(serial)
   end
+
+  private
+
+  def new_revocations(serials)
+    already_revoked = certificate_revocations.pluck(:serial)
+    (serials - already_revoked).map do |serial|
+      CertificateRevocation.new(serial: serial, certificate_authority: self)
```

Comment:
I'm in the process of making this a bit faster by not creating an ActiveModel object for each row.